tests/component/records/pandas/test_to_csv_options.py

```diff
@@ -0,0 +1,234 @@
+
+    def test_datetimeformattz(self) -> None:
+        known_failures = {
+            # We could probably support this once to_csv_options.py is
+            # less an if/else lookup and more of a function.
+            'YYYY-MM-DD HH12:MI AM',
+        }
+        expectations = {
+            'YYYY-MM-DD HH24:MI:SS': '%Y-%m-%d %H:%M:%S.%f',
+            'YYYY-MM-DD HH:MI:SS': '%Y-%m-%d %H:%M:%S.%f',
+            'MM/DD/YY HH24:MI': '%m/%d/%y %H:%M:%S.%f',
+        }
+        compatible_dateformat = {
+            'YYYY-MM-DD HH24:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH12:MI AM': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+        }
+        for datetimeformat in DATETIME_CASES:
+            records_format = DelimitedRecordsFormat(hints={
+                # Pandas doesn't consider dateformats to be separate
+                # from datetime/datetimetz formats, so they need to be
+                # consistent
+                'dateformat': compatible_dateformat[datetimeformat],
+                'datetimeformat': datetimeformat,
+                'datetimeformattz': datetimeformat,
+                'compression': None,
+            })
+            unhandled_hints = set(records_format.hints)
+            processing_instructions = ProcessingInstructions()
+            try:
+                options = pandas_to_csv_options(records_format,
+                                                unhandled_hints,
+                                                processing_instructions)
+            except NotImplementedError:
+                if datetimeformat in known_failures:
+                    continue
+                else:
+                    raise
+            self.assertEqual(options['date_format'], expectations[datetimeformat])
+
+            fileobj = io.StringIO(create_sample(datetimeformat))
+            df = pandas.DataFrame(data={'datetime':
+                                        [pandas.Timestamp(day=SAMPLE_DAY,
+                                                          month=SAMPLE_MONTH,
+                                                          year=SAMPLE_YEAR,
+                                                          hour=SAMPLE_HOUR,
+                                                          minute=SAMPLE_MINUTE,
+                                                          second=SAMPLE_SECOND)]},
+                                  columns=['datetime'])
+            df.to_csv(path_or_buf=fileobj,
+                      index=False,
+                      **options)
+            output = fileobj.getvalue()
+            # In reality this isn't used raw, as Pandas doesn't really
+            # try to handle lone dates or times.  Instead, we use
+            # prep_for_csv() to preconvert these Serieses into strings.
+            sample = create_sample(datetimeformat)
+            # TODO: Why are we including microsecond?  Is that in
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker